apps/frontend/src/libs/components/link/link.tsx

```diff
@@ -20,9 +26,26 @@ const Link: React.FC<Properties> = ({
 	to,
 	type,
 }: Properties) => {
+	const { hasUnsavedChanges } = useAppSelector(
+		({ unsavedChanges }) => unsavedChanges,
+	);
```

Comment:
I created a separate slice for this, since the project is navigated from the `Link` component, and the form that affects whether to open this `Popup` is located in `Settings`. Can you tell me how to find out about changes in the form inside the Link component?

Reply:
Check `useBlocker` from react-router-dom